### DISCUSSION TITLE
Extracting reliability branching scores

### DISCUSSION THREAD

**Discussion by cwfparsonson** [ds4dm/ml4co-competition#54] [Q&A] [answered]
Hi,

Is there a way to extract the reliability branching scores for each variable in an ecole environment, similar to `ecole.observation.StrongBranchingScores`, or does ecole only support extracting pseudocost and strong branching scores at the moment?

**Comment by gasse** [comment, marked as answer]
Hi @cwfparsonson ,

In Ecole currently we only have the strong branching scores and pseudocost scores available. I am not sure there is an easy way to extract reliability branching scores from SCIP unfortunately. Maybe @ambros-gleixner or @CGraczyk might have additional insight on how to extract something close using the PySCIPOpt API, but I am not sure SCIP computes or stores those scores outside of the [reliable pseudocost](https://www.scipopt.org/doc/html/branch__relpscost_8h.php) branching rule.

Best,
Maxime

**Reply by ambros-gleixner** [reply]
No, this is currently not possible, not even on the C level. With some effort it would be possible to write an API function to the branch_relpscost plugin on the C level, but this would certainly not be a method that just observes, but would also modify the internal SCIP statistics, pseudo costs, or even prune nodes as a result of strong branching.

**Reply by cwfparsonson** [reply]
Okay, thanks!
